server/src/main/java/org/opensearch/http/AbstractHttpServerTransport.java

```diff
@@ -352,17 +370,55 @@ protected void serverAcceptedChannel(HttpChannel httpChannel) {
      * @param httpChannel that received the http request
      */
     public void incomingRequest(final HttpRequest httpRequest, final HttpChannel httpChannel) {
-        handleIncomingRequest(httpRequest, httpChannel, httpRequest.getInboundException());
+        try (ThreadContext.StoredContext ignore = threadPool.getThreadContext().stashContext()) {
+            // TODO: Add support for parsing the otel incoming tracer.
+            final SpanScope httpRequestSpanScope = telemetryTracer.startSpan(
+                SPAN_NAME_INCOMING_HTTP_SERVER_REQUEST,
```

Comment:
Such names like SPAN_NAME_INCOMING_HTTP_SERVER_REQUEST (and other places) are not useful, the general convention is to have descriptive span naming, for example: `GET /_search` or `POST /_bulk `. When one looks into trace view, it is immediately clear what is going on.

Reply:
make sense.